src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/ApiSearchHelper.java

```diff
@@ -54,53 +64,42 @@ public void countRecentRepliesGiven(UserInteractions userInteractions, Date most
+                               .map(tweet -> this.getTwitterClient().getInitialTweet(tweet, true))
+                               .filter(initialTweet -> this.getUserId().equals(initialTweet.getAuthorId()))
+                               .peek(initialTweet -> LOGGER.info("."))
+                               .groupBy(ITweet::getId)
+                               .map(this::foldTweets);
+    }
 
-        for(ITweet tweet : tweetWithReplies){
-            if(this.isUserInList(tweet.getAuthorId())){
-                ITweet initialTweet = this.getTwitterClient().getInitialTweet(tweet, true);
-                if(this.getUserId().equals(initialTweet.getAuthorId())){
-                    System.out.print(".");
-                    // if the initial tweet id is not on the map, we add it
-                    if(!statusesAndAnswers.containsKey(initialTweet.getId())){
-                        statusesAndAnswers.put(initialTweet.getId(), new ArrayList<>());
-                    }
-                    if(!statusesAndAnswers.get(initialTweet.getId()).contains(tweet.getAuthorId())) {
-                        UserInteractions.UserInteraction userInteraction = userInteractions.get(tweet.getAuthorId());
-                        userInteraction.incrementNbRepliesReceived();
-                        statusesAndAnswers.get(initialTweet.getId()).add(tweet.getAuthorId());
-                        savedAnswers++;
-                    }
-                }
-            }
+    private Tuple2<String, TweetInteraction> foldTweets(String id, Stream<ITweet> tweets) {
+        return Tuple.of(id,
+                        tweets.foldLeft(new TweetInteraction(),
+                                        (interaction,  tweet) -> interaction.addAnswerer(tweet.getAuthorId())));
+    }
 
+    private Stream<ITweet> tweetsHavingReplies(boolean currentWeek) {
+        if(currentWeek){
+            val toDate = DateUtils.truncate(ConverterHelper.minutesBeforeNow(120), Calendar.HOUR);
+            return Stream.ofAll(this.getTwitterClient()
+                                    .searchForTweetsWithin7days("(to:"+this.getUserName()+" has:mentions)" + "OR (url:redtheone -is:retweet)",
+                                                                DateUtils.ceiling(DateUtils.addDays(toDate, -7),Calendar.HOUR),
+                                                                toDate));
         }
-        LOGGER.info("\n" + tweetWithReplies.size() + " replies to user found, " + savedAnswers + " saved");
+
+        val toDate = DateUtils.truncate(ConverterHelper.dayBeforeNow(7), Calendar.DAY_OF_MONTH);
+        return Stream.ofAll(this.getTwitterClient()
+                                .searchForTweetsWithin30days("to:" + this.getUserName() + " has:mentions",
+                                                             DateUtils.ceiling(DateUtils.addDays(toDate, -23), Calendar.DAY_OF_MONTH),
+                                                             toDate));
```

Comment:
Globally you should use `LocalDateTime` here, or `ZonedDateTime` if you expect your time to be zoned (to grab information at correct local times).